pulsator4droid/src/main/java/pl/bclogic/pulsator4droid/library/roundedrectangle/RoundedRectanglePulsatorLayout.java

```diff
@@ -0,0 +1,167 @@
+            mCanvas = new Canvas(mBitmap);
+        }
+
+        float width = getWidth();
+        float height = getHeight();
+
+        mCanvas.drawColor(Color.BLACK, Mode.CLEAR);
+        for (int i = 0; i < mCount; i++) {
+
+            float maskLeft = (pulseRectangle.right - pulseRectangle.left - mMaskWidth) / 2;
+            float maskTop = (pulseRectangle.bottom - pulseRectangle.top - mMaskHeight) / 2;
+
+            float progress = ((System.currentTimeMillis() - mStartTime) % (float) mDuration) / mDuration;
+            float offsetProgress = (1.0f / mCount * i + progress) % 1.0f;
+            float radius = height / 2;
+
+            float left = maskLeft - (maskLeft * offsetProgress);
+            float top = maskTop - (maskTop * offsetProgress);
+            float right = width - left;
+            float bottom = height - top;
+
+            mPaint.setAlpha((int) (ALPHA_MAX * (1 - offsetProgress)));
+            mCanvas.drawRoundRect(left, top, right, bottom, radius, radius, mPaint);
+        }
+
+        if (mMask != null) {
+            mCanvas.drawPath(mMask, mMaskPaint);
+        }
+
+        canvas.drawBitmap(mBitmap, 0, 0, mBitmapPaint);
+
+        if (isStarted()) {
+            invalidate();
+        }
+    }
+
+    @Override
+    protected PulseShape getPulseShape(Paint paint) {
+        return new PulseRoundedRectangle(paint);
+    }
+
+    @Override
+    public void start() {
+        if (mPulseShape instanceof PulseRoundedRectangle) {
+            mIsStarted = true;
+            mStartTime = System.currentTimeMillis();
+            invalidate();
+        }
+    }
+
+    @Override
+    public void stop() {
+        mIsStarted = false;
+    }
+
+    @Override
+    protected void reset() {
+        boolean isStarted = isStarted();
+
+        stop();
+        clear();
+
+        if (isStarted) {
+            start();
+        }
+    }
+
+    private void clear() {
+
+        mBitmap = null;
+        mCanvas = null;
+    }
+
+    /**
+     * The pulse animation will be clipped in a rounded rectangle of the given size with a given
+     * radius. Useful when we need transparency in the middle.
+     *
+     * @param width Width in pixels for the rounded rectangle masking out the animation in the
+     * center. Providing a value &lt;=0 disables the masking.
+     * @param height Height in pixels for the rounded rectangle masking out the animation in the
+     * center. Providing a value &lt;=0 disables the masking.
+     */
+    public void setCenterRoundedRectangleMask(int width, int height) {
+        if (mPulseShape instanceof PulseRoundedRectangle) {
+            if (width < 1 || height < 1) {
+                // disable the onDraw method (masking)
+                setWillNotDraw(true);
+                mMaskWidth = 0;
+                mMaskHeight = 0;
+                mMask = null;
+            } else {
+                // enable the onDraw method (masking)
+                setWillNotDraw(false);
```

Comment:
DRY on the three lines below

Reply:
I don't understand. How could this be made DRYer?

Reply:
setParams(int width, int height, boolean willNotDraw)?

Reply:
done